frontend/device/android/powerd.lua

```diff
@@ -6,22 +6,16 @@ local AndroidPowerD = BasePowerD:new{
     fl_max = 100,
 }
 
--- Let the footer know of the change
-local function broadcastLightChanges()
-    if package.loaded["ui/uimanager"] ~= nil then
-        local Event = require("ui/event")
-        local UIManager = require("ui/uimanager")
-        UIManager:broadcastEvent(Event:new("FrontlightStateChanged"))
-    end
-end
-
 function AndroidPowerD:frontlightIntensityHW()
     return math.floor(android.getScreenBrightness() / self.bright_diff * self.fl_max)
 end
 
 function AndroidPowerD:setIntensityHW(intensity)
     -- if frontlight switch was toggled of, turn it on
-    android:enableFrontlightSwitch()
+    if not self.is_fl_sw_off then
```

Comment:
typo? off != on

Reply:
Yes. This leads to a minimal screenflicker. Corrected